rfcs/2023-01-06-layout.md

```diff
@@ -0,0 +1,460 @@
+  paddingX="spacing.1"
+>
+  <Button>Button 1</Button>
+  <Button marginTop="spacing.1">Button 2</Button>
+  <Button marginTop="spacing.2">Button 3</Button>
+</Box>
+```
+
+Styling props will be the preferred choice for minor adjustments that can be done directly on the component without a need for creating extra `Box` wrapper _(which also keeps the DOM less nested and easier to read)_. Later on, we might want to expand the scope of these props to cover things beyond layout if need arises.
+
+As a tradeoff, styling props are pretty dangerous in terms of the flexibility they provide. They give power to customize any component to beyond recognition, for example you could customize the padding that comes on `Button`. In light of this capability we should support only a restricted set of properties to begin with _(which can't have adverse effect on look and feel of components)_.
+
+Styling props should be the preferred choice for minor adjustments only. You may note `padding*` is skipped since it can break existing padding of blade components. Supported restricted properties:
+
+| Category            | Props                                                                          | Notes                                                                                                                                                               |
+| ------------------- | ------------------------------------------------------------------------------ | ------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| Spacing             | `margin*`                                                                      | `marginTop`, `marginLeft`, `marginRight`, `marginBottom`, `marginX`, `marginY` (React Native `marginHorizontal`, `marginVertical` mapped to these)                  |
+| Flexbox             | `alignSelf`, `justifySelf`, `order`                                            | Only restricted set of properties applicable on flex children. Note: `justifySelf` doesn't work for flex layout but it can be used for block layout (web only).     |
+| Position            | `position`, `zIndex`, `top`, etc.                                              | [List of props](https://github.com/styled-system/styled-system/blob/master/docs/table.md#position) referenced from styled-system.                                   |
+| Grid **(web only)** | `gridColumn*`, `gridRow*`, `gridArea`, `justifySelf`, `alignSelf`, `placeSelf` | Only restricted set of properties applicable on [grid children](https://css-tricks.com/snippets/css/complete-guide-grid/#aa-properties-for-the-childrengrid-items). |
+| Layout              | None                                                                           |                                                                                                                                                                     |
```

Comment:
Seeing the [spec](https://github.com/styled-system/styled-system/blob/master/docs/table.md#layout) here, why no `display` or `verticalAlign`? 
- A switch b/w a `display: none` and a `display: block` on an event handler is a valid use case for component styling
- If I don't have a parent component or a `Box` that's a flexbox a dev will have to rely on knowing the height of the container and apply manual marginTop to centrally align, `verticalAlign` may come in handy here.

Reply:
For the style props on components I think we should skip these (`display` property will be supported on `Box` component). Reasoning:
- Having the ability to modify `display` property on presentational components can break styling (folks can pass `display` as `block` where the component's outer container might be `flex` styled and vice-versa)
- So for conditional rendering of presentational components users may render it as `condition ? <Alert /> : null`
- `vertical-align` would work only for table cells or `inline-*`, I don't think we've lot of presentational components that would fit this usage. For the table use case users may pass `vertical-align` to their `td` element 🤔

Reply:
Oh I didn't understand the display part. We are supporting it in Box right?

Reply:
Yup, we would definitely need `display` on `Box`. The tables have properties listed for both `Box` and other presentational components.